src/dataflow-types/src/client.rs

```diff
@@ -736,6 +736,13 @@ pub type LocalStorageClient = LocalClient<StorageCommand, StorageResponse>;
 /// A [`LocalClient`] for the compute layer.
 pub type LocalComputeClient = LocalClient<ComputeCommand, ComputeResponse>;
 
+/// Trait for clients that can be disconnected and reconnected.
+#[async_trait]
+pub trait Reconnect {
```

Comment:
This probably should inherit from `GenericClient`, yeah?

```suggestion
pub trait Reconnect<C, R>: GenericClient<C, R> {
```

Reply:
That doesn't gain you anything, but requires putting a bunch of random unnecessary trait bounds on e.g. the TcpClient impl (because its GenericClient impl has those bounds...)

I can add it if you want, seems like stylistic preference though.

Reply:
Yeah I guess it doesn't much matter since the definition of `Reconnect` doesn't need to reference anything about `GenericClient`.